### ISSUE TITLE
Add support of MarkDown syntax ?

### ISSUE THREAD

**Issue by LudovicRousseau** [aykit/MyOwnNotes#103]
It would be nice to be able to render the notes text as markdown.

On the server side it is possible to use owncloud-markdown https://github.com/icewind1991/owncloud-markdown to edit text as MarkDown
The problem is the server Notes application and the Android Notes application can only use .txt files (and not .md files).
1. allow the Notes applications to use the .md extension. The files are still pure text files
2. allow the use of an external renderer on Android to render the notes as MarkDown

The best would be to make myownnotes-android render the MarkDown by itself. But I understand that is work.

**Comment by ramschmaerchen** [comment]
Hej @LudovicRousseau,
in my opinion it is all about the question what notes are / should be. We thought of notes as short text paragraphs, only for note taking. Markdown is not necessary in this scenario.

On the other hand, one could think about notes being extensive texts about subjects. Markdown would be quite interesting in this scenario.

We aykit were aiming at the first scenario. That is why we chose the server app notes, as it was similar in it's approach. Additionally, as far as I know, the notes app was able to do markdown once, but was dumbed down. I guess for similar reasons as mentioned above.

I would say I am a bit opposed to the idea of a featurefull notes app as I really like a small, dumb application like myownnotes but I guess it is a topic that should be discussed.

However, please bear in mind that the android version is only a gui for the server-side application. I therefore suggest you open feature requests like those at the repository of the server-side application.

**Comment by eephyne** [comment]
I can understand the will to make the app simplier but I personally found that markdown is really convenient for note taking: it's simple, easy, and understandable (even if preview is better than just plain text).

The upper reference suggesting that's planned in the note app so I guess that's just a matter of time.

That's just my opinion and I understand well that represent a lot of work, I just wanted to say that I'd like that too.

**Comment by Fehlersturm** [comment]
Just so i und erst an this. Ehen would you want to See a rendered view of
your notes written in MD. Or are you looking for Editor features like auto
complete and highlighting?

**Comment by eephyne** [comment]
Woo  look like mobile auto correction :)

Both would be awesome,  well I personally don't search auto complete and highlight but more for easy access to MD characters ()[]~*-+! .

Primarily I think render view is more important since editor is already present

**Comment by Fehlersturm** [comment]
Jupp, phone tried to germanify my English.
So you will need to use a different keyboard for special characters.

And I still don't quite get how the rendered view would work. Would there
be a edit button? So default view would be: try to render MD/rsT. then push
edit for text editor? Because that wouldn't need a server side
implementation at all.   Cause server is just a notes folder with text
files. MD is text files.

**Comment by eephyne** [comment]
You can see in some apps for mobile (eg: writeily pro on android) there is just a line above the keyboard that contains all the character needed for easy access.
Screenshot :
![](http://i.imgur.com/c1hHirN.jpg)
The second bar is not related to the virtual keyboard.

For render yes a preview button is generally what's used.
(In my opinion a swipe movement to access the render could be really nice)

And yes you need to modify note app because note app only read text files, it ignore .MD or .mkd files. (So myownnote doesn't see them)
Of course myownnote can ignore that and threat txt as md but I think that would be proper if both apps detect MD or txt file.

**Comment by maltefiala** [comment]
Sorry, won't fix.

**Comment by ramschmaerchen** [comment]
owncloud/notes will support markdown with the release of owncloud 8.1

I still think it's highly impractical on a mobile device. Typing in === (( )) [[]] characters on a mobile device does not look very convenient to me.

As the parent app will support it, we'll have to think about what to do. Will definitely not be part of Version 2.0

**Comment by ramschmaerchen** [comment]
Forgot: The new lines make the screen practically disappear. It looks like wasting a lot of my expensive display space. If we implement it, it should be done right. No idea what right would look like, however...

**Comment by LudovicRousseau** [comment]
I do not use MyOwnNotes anymore. I use a normal text file (using a ``.md`` extension) synchronised using the standard OwnCloud Android application. I do not have a markdown rendering on Android but that is not that important.

So feel free to close this bug.

**Comment by ramschmaerchen** [comment]
Ok, good to know. Well, we will keep it open as we'll have to think about this soon. Thanks for your feedback.

**Comment by edent** [comment]
Just another request for this feature.  I use this on my tablet - so there's plenty of space for viewing files. It's also easy enough to type a `*` or `[`.

[OT] @LudovicRousseau how are you syncing a folder to OwnCloud using the official app? I can tell a folder to sync, but it doesn't seem to detect when I save a new file locally using, say, Writeily.

**Comment by LudovicRousseau** [comment]
@edent As I wrote in my previous comment I do not use MyOwnNotes anymore.

**Comment by LEDfan** [comment]
@ramschmaerchen 
> We aykit were aiming at the first scenario. That is why we chose the server app notes, as it was similar in it's approach. Additionally, as far as I know, the notes app was able to do markdown once, but was dumbed down. I guess for similar reasons as mentioned above.

> I would say I am a bit opposed to the idea of a featurefull notes app as I really like a small, dumb application like myownnotes but I guess it is a topic that should be discussed.

I agree with you, that a note app should have just an editing area and a new note button.  (a desktop app could have some more features as long it is not cluttered) But not everyone agrees with this, and that's why I think you should support the extensions the notes app on ownCloud supports. (https://github.com/owncloud/notes/pull/161) Note that with support I don't mean to render, highlight etc of markdown files. The notes app of ownCloud also doesn't support this AFAIK. I also don't think this is necessary for an andoird app. 

TL:DR make it possible to open `.md` files, but just act like they are `.txt` files.

Thanks in advance!

**Comment by LEDfan** [comment]
Mmm, it looks like it's already supported to open and edit markdown files, sorry then for your time :smiley:

**Comment by maltefiala** [comment]
MyOwnNotes 2 is live. Complete rewrite, issue closed
